packages/babel-helpers/src/helpers.js

```diff
@@ -413,6 +416,29 @@ helpers.interopRequireWildcard = template(`
   })
 `);
 
+helpers.specRequireInterop = template(`
+  (function (obj) {
+    if (obj && obj.__esModule) {
+      return obj;
+    } else {
+      var newObj = Object.create(null, {
```

Comment:
Hm, it could potentially work with a `spec` + `loose` version that disables the commonjs shadowing and uses similar export logic to the non-`spec` version...